SemanticUI/Modules/Dropdown.elm

```diff
@@ -206,15 +220,43 @@ dropdown config =
         }
 
 
+{-| A button dropdown variation
+-}
+button :
+    { config | button : Button.Config msg, drawerState : DrawerState, identifier : String, onToggle : DrawerState -> msg }
+    -> Dropdown msg
+button config =
+    let
+        (Dropdown ordinaryConfig) =
+            dropdown config
+    in
+    Dropdown { ordinaryConfig | variation = Button config.button }
+
+
 {-| Render a dropdown with the provided layout
 -}
 toCustomHtml : (Builder msg -> Html msg) -> Dropdown msg -> Html msg
 toCustomHtml layout (Dropdown config) =
-    layout
-        { toToggle = toToggle config
-        , toDropdown = toRoot config
-        , drawer = drawer config
-        }
+    case config.variation of
+        Ordinary ->
+            layout
+                { toToggle = toToggle config
+                , toDropdown = toRoot config
+                , drawer = drawer config
+                }
+
+        Button but ->
+            layout
+                { toToggle = toToggle config
+                , toDropdown =
+                    \element attrs ->
+                        Button.viewAs
+                            (toRoot config element
+                                |> HtmlBuilder.prependAttributes attrs
+                            )
+                            but
```

Comment:
This is getting to the point where I'm getting pretty confused about the ordering of attributes.

I'd be happy to take suggestions on how to simplify. I think I'll need to review exactly how the ordering ends up